spec/lib/hypgen/planner_spec.rb

```diff
@@ -13,12 +12,54 @@
       [
         { init_conf_tmp_id: 7, params: { experimentId: @experimentId, dap_token: Hypgen.config.dap_token } },
       ]}
+    allow(workflow).to receive(:start_time) {"2014-11-19T09:45:42.025Z"}
+    allow(workflow).to receive(:end_time) {"2014-11-20T09:45:42.025Z"}
     planner = Hypgen::Planner.new(workflow)
     setup     = planner.setup
 
     puts setup[0][:vms]
 
-    expect(setup[0][:vms].count).to eq(3)
+    expect(setup[0][:vms].count).to eq(2)
+  end
+
+  it 'should read start and end time from workflow' do
+    workflow = double("workflow")
+    allow(workflow).to receive(:deadline) {60}
+
+    allow(workflow).to receive(:start_time) {"2014-11-19T09:45:42.025Z"}
```

Comment:
Space missing inside {.<br>Space missing inside }.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.